kedro/config/omegaconf_config.py

```diff
@@ -141,8 +141,17 @@ def __init__(  # noqa: too-many-arguments
             env=env,
             runtime_params=runtime_params,
         )
+        try:
+            self._globals = self["globals"]
+        except MissingConfigException:
+            self._globals = {}
 
-    def __getitem__(self, key) -> dict[str, Any]:
+    def __setitem__(self, key, value):
+        if key == "globals":
+            self._globals = value
+        super().__setitem__(key, value)
```

Comment:
```suggestion
    def __setitem__(self, key, value):
        if key == "globals":
        # Update the cached value at self._globals since it is used by the globals resolver
            self._globals = value
        super().__setitem__(key, value)
```